collect_app/src/test/java/org/odk/collect/android/projects/ProjectsRepositoryTest.kt

```diff
@@ -0,0 +1,25 @@
+package org.odk.collect.android.projects
+
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.`is`
+import org.junit.Before
+import org.junit.Test
+
+abstract class ProjectsRepositoryTest {
```

Comment:
Given we have implementation for them, it would be good to have tests for `get` and `delete`. We also want to check that each Project gets its own ID. IMO, this doesn't need to do anything to mathematically validate uniqueness. There's a lost example in `InstancesRepositoryTest#save_addsUniqueId`. 

My tip for (especially but not just) these kinds of objects and these kinds of tests is to use strict TDD and not write any code in the object until there is a failing test. That help keeps the coverage really tight and means there is documentation for every piece of behaviour we expect from the interface. Most importantly, I find I have an easier time working out what tests to write. I completely understand that TDD can be a weird way to work for a lot of people though so happy to pair on it or run through examples.

Reply:
I added more tests.